pulp_rpm/app/migrations/0017_merge_advisory_collections.py

```diff
@@ -0,0 +1,45 @@
+from django.db import migrations
+
+def do_clone(advisory, in_collection):
+    # We're cloning collection - find its updatecollectionpackages
+    uc_packages = in_collection.packages.all()
+    # break the advisory/collection link
+    in_collection.update_record.remove(advisory)
+    # create a new copy of the collection and link it to the advisory
+    new_collection = in_collection
+    new_collection.pulp_id = None
+    new_collection.pk = None
+    new_collection.save()
+    # need to have an id before we can build the m2m relation
+    new_collection.update_record.add(advisory)
+    new_collection.save()
+    for a_package in uc_packages:
+        # create copies of the package list and link to new collection
+        a_package.pulp_id = None
+        a_package.pk = None
+        a_package.update_collection = new_collection
+        a_package.save()
+
+def clone_reused_update_collections(apps, schema):
+    # Find UpdateCollections that point to multiple UpdateRecords
+    # For all but the first one, create a clone
+    UpdateCollection = apps.get_model("rpm", "UpdateCollection")
+    collections = UpdateCollection.objects.all().order_by('pulp_id')
```

Comment:
Why do they need to be ordered?

Is it safe to do objects.all()? There can be a noticeable amount of collections present in pulp, I think. Do we want to load them all into memory. They are not big , to be fair.
Alternatively, does it makes sense to select only ones with multiple relations? maybe by using annotate?

Reply:
Good catch RE ordered, that was left over from an initial effort. I can (and will) make this an iterator, which shouldn't load every collection into memory at once (I think) - but limiting to "that point to more than one advisory" to begin with is a fine idea, will investigate that change as well.

Reply:
FWIW, the actual sorting is done by the database, and that usually doesn't do the sorting in memory unless it will easily fit.  I think in this case it would actually use the index  (maybe not, idk w/ uuids) but otherwise it would happen on disk.

Reply:
Modified to do annotate/filter, so we only actually deal with collections whose num_advisories >= 2. Also modified to do bulk-save on UpdateCollectionPackage.